eth/downloader/queue.go

```diff
@@ -101,11 +101,14 @@ type queue struct {
 	resultCache  []*fetchResult // Downloaded but not yet delivered fetch results
 	resultOffset uint64         // Offset of the first cached fetch result in the block chain
 
-	lock sync.RWMutex
+	lock   *sync.Mutex
```

Comment:
Why change it to a plain mutex? What's wrong with allowing multiple goroutiens to read the state concurrently?